tests/src/test/scala/akka/kafka/scaladsl/IntegrationSpec.scala

```diff
@@ -373,9 +373,10 @@ class IntegrationSpec extends SpecBase with TestcontainersKafkaLike with Inside
         .run()
 
       // Wait a tiny bit to avoid a race on "not yet initialized: only setHandler is allowed in GraphStageLogic constructor"
```

Comment:
I believe you need to keep the sleep to avoid the race (it rarely shows).

Reply:
Something was triggering the race condition frequently after the upgrade. I played with different intervals, but low numbers didn't seem to work consistently. I'll take another look.

Reply:
I'll keep both.